samples/integration/vcpkg-all-smoke/src/main.cpp

```diff
@@ -67,7 +67,7 @@ int main()
 
     std::cout << "Successfully Created the Clients";
   }
-  catch (std::exception exception)
+  catch (std::exception const& exception)
```

Comment:
Constructors can and should throw. You either get correctly constructed object, or you get an exception during creation. Desctructors should not throw - because a destructor may be invoked when stack unwinding is in progress, and that unwinding happens when another exception is thrown. If you throw another exception during stack unwinding, C++ runtime will call `termainate()` which is pretty bad. Plus, when you create an it throws - you normally know how to handle the error. But if it is in a destructor - the code that holds the object is lokely does not know what to do with it. Imagine an object that is held in a shared pointer, as a pretty abstract interface - let's say some reader that may read from network, user input or file, and you happen to be the last out of many who holds it. And then it throws a network exception during destruction - you have no idea what to do with it.